Formula/node@14.rb

```diff
@@ -22,21 +23,24 @@ class NodeAT14 < Formula
   keg_only :versioned_formula
 
   depends_on "pkg-config" => :build
-  depends_on "python@3.10" => :build
   depends_on "brotli"
   depends_on "c-ares"
   depends_on "icu4c"
   depends_on "libnghttp2"
   depends_on "libuv"
   depends_on "openssl@1.1"
 
-  uses_from_macos "python"
   uses_from_macos "zlib"
 
   on_macos do
+    depends_on "python@3.10" => [:build, :test]
     depends_on "macos-term-size"
   end
 
+  on_linux do
+    depends_on "python@3.10"
```

Comment:
Not sure of best way to handle Python dependency.

At least on Linux, only using Python 3.10 seems better than weird build with Python 3.10 but run with Python 3.9.

On macOS, most scripts seem to work with system Python 3.8 / Python 2.7, but there is one gyp script used in test that runs `#!/usr/bin/env python` which doesn't work on Monterey. For now, just added as test dependency to handle this.

I was originally looking at an env script pointing to Python's `libexec/bin`, but didn't want to risk breaking something, so just going with test block modification. If macOS Monterey `node@14` users actually see issues without `python` on PATH, then can look into a workaround at that time.